solidity/test/data/redemption.ts

```diff
@@ -0,0 +1,179 @@
+import { BytesLike } from "@ethersproject/bytes"
+import { BigNumber, BigNumberish } from "ethers"
+
+/**
+ * Represents a set of data used for given redemption scenario.
+ */
+export interface RedemptionTestData {
+  /**
+   * Wallet that makes the redemption transaction.
+   */
+  wallet: {
+    pubKeyHash: BytesLike
+    state: BigNumberish
+    pendingRedemptionsValue: BigNumberish
+  }
+
+  /**
+   * Redemption requests that are handled by the redemption transaction.
+   */
+  redemptionRequests: {
+    redeemer: string
+    redeemerOutputScript: BytesLike
+    amount: BigNumberish
+  }[]
+
+  /**
+   * Main UTXO data which are used as `mainUtxo` parameter during
+   * `submitRedemptionProof` function call. Main UTXO must exist for given
+   * wallet in order to make the redemption proof possible
+   */
+  mainUtxo: {
+    txHash: BytesLike
+    txOutputIndex: number
+    txOutputValue: BigNumberish
+  }
+
+  /**
+   * Redemption transaction data passed as `redemptionTx` parameter during
+   * `submitRedemptionProof`function call.
+   */
+  redemptionTx: {
+    hash: BytesLike
+    version: BytesLike
+    inputVector: BytesLike
+    outputVector: BytesLike
+    locktime: BytesLike
+  }
+
+  /**
+   * Redemption proof data passed as `redemptionProof` parameter during
+   * `submitRedemptionProof` function call.
+   */
+  redemptionProof: {
+    merkleProof: BytesLike
+    txIndexInBlock: BigNumberish
+    bitcoinHeaders: BytesLike
+  }
+
+  /**
+   * Chain difficulty which was in force at the moment of Bitcoin transaction
+   * execution. It is used to mock the difficulty provided by `Relay` contract
+   * with a correct value thus making proof validation possible.
+   */
+  chainDifficulty: number
+}
+
+/**
+ * Container for TBTC balance change during redemption test scenario.
+ */
+export interface RedemptionBalanceChange {
+  beforeProof: BigNumber
+  afterProof: BigNumber
+}
+
+/**
+ * `MultiplePendingRequestedRedemptionsWithChange` test data represents a
+ * redemption with the following properties:
+ * - 5 redemption requests
+ * - Redemption dust threshold is 1000000 satoshi
+ * - Treasury fee for each request is 100000 satoshi
+ * - Maximum transaction fee for each request is 1000 satoshi
+ * - Total requested amount for all requests is 6934567 satoshi
+ * - Total treasury fee for all requests is 500000 satoshi
+ * - Total redeemable amount for all requests is 6434567 satoshi
+ * - Maximum total transaction fee is 5000 satoshi
+ */
+export const MultiplePendingRequestedRedemptionsWithChange: RedemptionTestData =
+  {
+    wallet: {
+      pubKeyHash: "0x7ac2d9378a1c47e589dfb8095ca95ed2140d2726",
+      state: 1,
+      pendingRedemptionsValue: 0,
+    },
+
+    redemptionRequests: [
+      {
+        redeemer: "0x7F62CddE8A86328d63B9517BC70B255017f25EEa",
+        redeemerOutputScript:
+          "0x1976a9142cd680318747b720d67bf4246eb7403b476adb3488ac", // P2PKH
```

Comment:
Can we add a comment with address so that the output script can be matched with UTXO from https://live.blockcypher.com/btc-testnet/tx/05dabb0291c0a6aa522de5ded5cb6d14ee2159e7ff109d3ef0f21de128b56b94/ ?